pymc_experimental/tests/test_marginal_model.py

```diff
@@ -411,3 +415,14 @@ def test_marginalized_transforms(transform, expected_warning):
             transform_name = transform.name
         assert f"sigma_{transform_name}__" in ip
     np.testing.assert_allclose(m.compile_logp()(ip), m_ref.compile_logp()(ip))
+
+
+def test_is_conditional_dependent_static_shape():
+    """Test that we don't consider dependencies through "constant" shape Ops"""
+    x1 = pt.matrix("x1", shape=(None, 5))
+    y1 = pt.random.normal(size=pt.shape(x1))
+    assert is_conditional_dependent(y1, x1, [x1, y1])
```

Comment:
I would have guessed this case would fail, because there is a `None` in the shape (line 256)?